clients/src/main/java/org/apache/kafka/clients/producer/TransactionState.java

```diff
@@ -116,6 +116,7 @@ public synchronized Integer sequenceNumber(TopicPartition topicPartition) {
 
     public synchronized void incrementSequenceNumber(TopicPartition topicPartition, int increment) {
         if (!sequenceNumbers.containsKey(topicPartition)) {
+            // TODO: Should this be an illegal state or can this case actually happen?
```

Comment:
I don't think it should ever happen, might just throw an `IllegalStateException`